docs/DeveloperGuide.md

```diff
@@ -1,34 +1,357 @@
+
+![](Images/Image1.PNG)
+
+### 2.3 Importing into IntelliJ [optional]
+
+In case you prefer to use IDEs to develop. 
+IntelliJ IDEA community edition is a popular free choice. 
+After installing a version of it you can import it using the “Get from Version Control” option below and clone from our [repository](https://github.com/AY2021S1-CS2113T-W11-3/tp.).
+
+![](Images/Image2.PNG)
+
+## 3. Design
+
+This section will elaborate on the architecture and component design of CheatLogs.
+
+### 3.1. Architecture
+
+The image below illustrates the high-level design of CheatLogs.
+
+![](Images/Image3.PNG)
+
+Image 1: General Architecture of CheatLogs
+
+CheatLogs is split into 5 major components, each handling distinct features of the application. The components and a brief description of them is listed below.
+* `UI`: The user interface of the app.
+* `CheatSheet`: The structure of each cheat sheet
+* `CheatSheetList`: A collection of every cheat sheet.
+* `Parser`: Builds a data structure based on user inputs.
+* `Command`: An encapsulation of data and methods to execute each command
+* `Storage`: Updates application data based on relevant external files.
+
+The UML diagram below illustrates an extensive version of the various classes present in CheatLogs as well as their interactions with each other.
+
+![](Images/Image4.PNG)
+
+Image 2: In-Depth Architecture of CheatLogs
+
+For each component, it can be further split into different classes which have a unique responsibility. They will be further elaborated upon in the next section.
+
+
+### 3.2. Components
+
+This application is divided into five different components. 
+Each component has a unique function and contributes to the functionality of this application.
+
+#### 3.2.1. User Interface (Abner)
+
+This component handles interactions with the user and manages the input and output of the programme. 
+
+![](Images/Image5.PNG)
+
+Image 3: User Session class fields and methods
+
+One of the classes is UserSession which contains the main loop of the program. 
+Every loop it reads and parses the user input to get a Command object which encapsulates all the necessary information needed to execute the command. 
+After execution, it handles potential exceptions thrown.
+
+
+A single instance of common objects are usually created in UserSession, such as Ui and Printer objects. 
+These common objects are injected into other objects that need them via the class constructor instead of static methods.
+The common objects include the Ui and  Printer helper classes which provide an organized way to read and write data. 
+Most of the programmes’ output is made via calls to the same common printer object.
+
+#### 3.2.2. Command Parser (Brandon)
+
+This component would parse the user input to produce useful information which would be used to construct a Command and executed the command. 
+
+![](Images/Image6.PNG)
+
+Image 4: Interaction between components to parse and execute command
+
+![](Images/Image7.PNG)
```

Comment:
![image](https://user-images.githubusercontent.com/28430109/97656068-821e1b00-1aa1-11eb-8b64-dcbc34292470.png)

Perhaps you can include the dotted return line?